docs/design.md

```diff
@@ -0,0 +1,52 @@
+# Cyclone Design
+
+Cyclone is a powerful workflow engine and end-to-end pipeline solution implemented with native Kubernetes resources. 
+Cyclone is architectured with a low-level workflow engine that is application agnostic, offering capabilities like
+workflow DAG scheduling, resource lifecycle management.
+
+## Architecture
+
+![Cyclone arch](images/cyclone-arch.png)
+
+### Components
+
+* Cyclone Web: UI of Cyclone helps users to easily use Cyclone.
+* Cyclone Server: API server of Cyclone provides a set of RESTful APIs to operate Cyclone resources.
+* Workflow Controller: Controller of Cyclone workflow engine, controls execution of stages in workflows.
+
+### Kubernetes CRDs
+
+* Stage: Minimum executable unit for workflow, it does some work on input resources and output results.
+* Resource: Input or output of stages.
```

Comment:
`Input or output of stages` is not accurate, it includes both`resource` and `artifacts`. I think here we should explain what resource is, of course, it can be used as _Input or output of stages_

Reply:
What about?
```
Resource is the data used by stages, it can be the input or output of stages.
```

Reply:
`Resource is the data used by stages as input or output, for example, git repository in SCM or docker images in registry`